### PR TITLE
feat(@desktop/wallet): Add Select Input and  output params to swap modal

### LINKED ISSUE #14826: Select output params
https://www.notion.so/Select-output-parameters-ce74bb4ee5fe43f9923476a26e8ff130
https://www.figma.com/design/TS0eQX9dAZXqZtELiwKIoK/Swap---Milestone-1?node-id=3406-234619&t=eTvhBjNa4YYvdRU7-0

- Select output asset
Fill selector with tokens with balance >0 and remaining tokens from the Token List
- Enter output amount
Input validation

### PR DESCRIPTION
**Pull request by Khushboo-dev-cpp**
fixes #14826, #14825

### What does the PR do

This PR uses the SwapInputPanel and connects it to the backend. it tried to cover all the flows of Swap M1 designs related to input and output fields that can be supported and added TODO is places there are bugs such as the ones listed here https://github.com/status-im/status-desktop/issues/15162

Have written UTS for all possible scenarios but may need more as we find more corner cases. some tests are comented because of the issue mentioned above. 

US's covered
https://www.notion.so/Select-input-parameters-56c5f81573d145f6b079678cdaaa362c?pvs=4 and 
https://www.notion.so/Select-output-parameters-ce74bb4ee5fe43f9923476a26e8ff130?pvs=4

### Affected areas

SwapModal, SwapInputPanel

### StatusQ checklist

- [ ] add documentation if necessary (new component, new feature)
- [ ] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [ ] test changes in both light and dark theme?

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

https://github.com/status-im/status-desktop/assets/60327365/217f1a27-5b10-4077-be7d-9b2209c9030d

https://github.com/status-im/status-desktop/assets/60327365/dbc95326-e842-4491-808a-f8969fa111e3

### CODE REVIEW

File: ui/app/AppLayouts/Wallet/popups/swap/SwapModal.qml [hunk 1]
```diff
@@ -185,10 +218,17 @@ StatusDialog {
...
             Layout.topMargin: Style.current.smallPadding
-            text: qsTr("An error has occured, please try again")
+            text: {
+                if (payPanel.amountEnteredGreaterThanBalance)   {
+                    return qsTr("Insufficient funds for swap")
+                }
+                return qsTr("An error has occured, please try again")
+            }
+            buttonText: payPanel.amountEnteredGreaterThanBalance ? qsTr("Buy crypto"): ""
+            onButtonClicked: console.warn("TODO::: add buy crypto flow")
```
Comment (caybro) [inline ui/app/AppLayouts/Wallet/popups/swap/SwapModal.qml:231]: Don't we have already a "Buy crypto" popup?

```suggestion
            onButtonClicked: Global.openBuyCryptoModalRequested()
```

We do :)

**Review by caybro — APPROVED** [review]
Looks good!

File: ui/app/AppLayouts/Wallet/popups/swap/SwapModal.qml (hunk 1 above)
Comment (Khushboo-dev-cpp) [reply ui/app/AppLayouts/Wallet/popups/swap/SwapModal.qml]: good point, done!

File: ui/imports/shared/popups/send/views/AmountToSend.qml [hunk 2]
```diff
@@ -95,8 +95,11 @@ ColumnLayout {
                         d.cryptoValueToSend, root.multiplierIndex).toString()
         }
 
-        readonly property string zeroString:
-            LocaleUtils.numberToLocaleString(0, 2, topAmountToSendInput.locale)
+	        // Crypto value should be represented by 0 and fiat with 0.00
```
Comment (dlipicar) [inline ui/imports/shared/popups/send/views/AmountToSend.qml:98]: bad indentation?

**Review by dlipicar — APPROVED** [review]

File: storybook/qmlTests/tests/tst_SwapModal.qml [hunk 3]
```diff
@@ -599,6 +635,497 @@ Item {
...
+            compare(maxTagButton.text, qsTr("Max. %1").arg(root.swapAdaptor.currencyStore.formatCurrencyAmount(Math.trunc(WalletUtils.calculateMaxSafeSendAmount(expectedToken.currentBalance, expectedToken.symbol)*100)/100, expectedToken.symbol, {noSymbol: true})))
+            compare(payPanel.selectedHoldingId, expectedToken.symbol)
+            compare(payPanel.cryptoValue, 0)
+            compare(payPanel.cryptoValueRaw, SQUtils.AmountsArithmetic.fromNumber("0", expectedToken.decimals).toString())
+            verify(!payPanel.cryptoValueValid)
+
+            closeAndVerfyModal()
+        }
+
+        function test_modal_pay_input_switching_networks() {
```
Comment (Khushboo-dev-cpp) [inline storybook/qmlTests/tests/tst_SwapModal.qml:840]: on switching test networks the form should be reset!! needs to be changed to that behaviour https://www.figma.com/design/TS0eQX9dAZXqZtELiwKIoK/Swap---Milestone-1?node-id=3410-271335&m=dev
